### ISSUE TITLE
Training with a different loss (improvement)

### ISSUE THREAD

**Issue by Atheer2104** [Atheer2104/dd2424-CNN-classificaiton-project#15]
The cross-entropy loss is not the only loss that can be used for training a network. One 
alternative approach is to use supervised contrastive learning to pre-train the network. The 
idea was introduced in [Supervised Contrastive Learning](https://proceedings.neurips.cc/paper/2020/file/d89a66c7c80a29b1bdbab0f2a1a94af8-Paper.pdf) by Khosla et al, NeurIPS, 2020. 
Here the goal is learn a vector representation of images such that images from the same 
class have similar representations while those from other classes are dissimilar. To make your 
computations feasible you will probably have to make some compromises on batch sizes and 
how much data augmentation you apply. You should just investigate the loss Lout as this 
appear to consistently produce better results than the Lin loss. Once you have trained such 
a network then you can use this feature representation and train a small MLP with cross- 
entropy training to perform image classification. You should then compare performance the 
performance you achieve to that of a similarly sized network trained purely with cross-entropy.
